server/libs/domain/src/media/media.service.ts

```diff
@@ -151,32 +153,45 @@ export class MediaService {
     }
   }
 
-  private getLongestStream(streams: VideoStreamInfo[]): VideoStreamInfo | null {
-    return streams
-      .filter((stream) => stream.codecType === 'video')
-      .sort((stream1, stream2) => stream2.frameCount - stream1.frameCount)[0];
+  private getMainVideoStream(streams: VideoStreamInfo[]): VideoStreamInfo | null {
+    return streams.sort((stream1, stream2) => stream2.frameCount - stream1.frameCount)[0];
   }
 
-  private isTranscodeRequired(stream: VideoStreamInfo, ffmpegConfig: SystemConfigFFmpegDto): boolean {
-    if (!stream.height || !stream.width) {
+  private getMainAudioStream(streams: AudioStreamInfo[]): AudioStreamInfo | null {
+    return streams[0];
+  }
```

Comment:
Uploading a file with multiple audio or video streams is probably not that common, but I noticed we select the video stream with the highest frame count and for audio the first stream. Is there any reason behind this? The selected streams are also not passed on to ffmpeg which uses it's own [stream selection](https://ffmpeg.org/ffmpeg.html#Automatic-stream-selection) algorithm.

Reply:
Improving the actual commands sent to ffmpeg is further down my list, we can look at passing the stream identifier directly into ffmpeg at a later date. The reason for just selecting the first audio stream is simply that there isn't really any metadata available to easily identify the "best" stream. I thought about doing it based on bitrate of the stream, but that is actually only metadata added by the original encoder rather than being calculated by ffprobe at the time, so it can be wrong.

Reply:
Was just curious about the different approach between them. FFmpeg uses the stream with most audio channels, but since this is such an edge case it can definitely be addressed later on.